lib/NormalModule.js

```diff
@@ -780,6 +784,31 @@ class NormalModule extends Module {
 	 * @returns {Source} the created source
 	 */
 	createSource(context, content, sourceMap, associatedObjectForCache) {
+		// If this module exists in a layer, try to reuse the value for the
+		// source string so it is not duplicated in multiple modules.
+		if (this.layer && this.userRequest) {
```

Comment:
```suggestion
		if (this.layer && this.request) {
```

Reply:
You need to use `userRequest` since at least in Next.js App Router, the values of `request` will be different for the same file across the different layers.

Reply:
Could you use `resource` or `resourcePath`?

Reply:
`resource` also works.